### ISSUE TITLE
Name conflict in freshly generated .test.tsx file

### ISSUE THREAD

**Issue by BudWhiteStudying** [arminbro/generate-react-cli#127]
**Describe the bug**

When a new Typescript component (say `HelloWorld`) is generated, complete with a test file (`.test.tsx`), the file reports the following error: `Block-scoped variable 'HelloWorld' used before its declaration.`

**To Reproduce**

1. Generate a brand new `Next.js` React project using Typescript, add the necessary dependencies
    - `npx create-next-app@14.2.16 react-demo-app` (use default yes/no answers for every prompt)
    - `cd react-demo-app`
    - `npm install --save-dev styled-components`
    - `npm install --save-dev @types/react @types/react-dom @types/jest`
    - `npm install --save-dev jest npm @swc/core @swc/jest jest-environment-jsdom`
    - `npm install --save-dev @testing-library/react @testing-library/dom @testing-library/jest-dom @testing-library/user-event`

2. Set up testing config
    - create a `jest.config.js` file with the following content:
    ```javascript
    module.exports = {
        transform: {
          '^.+\\.(t|j)sx?$': '@swc/jest',
        },
        testEnvironment : "jsdom"
      }
    ```
    - update the `package.json` file by adding `"test": "jest"` to the `scripts` list property

3. Generate a `HelloWorld` component with `generate-react-cli`, using default yes/no answers for every prompt:

    ```bash
    npx generate-react-cli component HelloWorld
    ```

4. Navigate to the `components/HelloWorld` directory that was generated by the command, inspect the `HelloWorld.test.tsx` file within

5. See syntax/compilation error within the file

**Expected behavior**

No syntax errors in the freshly generated file.

**Machine (please complete the following information):**

 - OS: MacOS Sonoma
 - Node Version: `v18.19.1`

**Additional context**

Looking at [componentTestTestingLibraryTemplate.js](https://github.com/arminbro/generate-react-cli/blob/master/src/templates/component/componentTestTestingLibraryTemplate.js), it seems to me that the same placeholder is used for the import at line 4 and for the variable definition at line 10. I believe that's what is causing the issue.

In fact, if I rename the variable to something else e.g. `helloWorld` (lowercase `h` instead of uppercase) the error goes away. Indeed, the error message said `Block-scoped variable 'HelloWorld' used before its declaration.` so I guess it was referring to the declaration at line 10.

If that's OK with you, I would open a PR changing lines 10-12 to

```javascript
    const templateName = screen.getByTestId('templatename');

    expect(templateName).toBeInTheDocument();
```

**Comment by BudWhiteStudying** [comment]
Anyway, the issue can be circumvented by defining a custom test template:

```javascript
//component-templates/TemplateName.test.js
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import templatename from './templatename';

describe('<templatename />', () => {
  test('it should mount', () => {
    render(<templatename />);

    const templateName = screen.getByTestId('templatename');

    expect(templateName).toBeInTheDocument();
  });
});
```

```json
{
  "usesTypeScript": true,
  "usesStyledComponents": true,
  "testLibrary": "Testing Library",
  "component": {
    "default": {
      "customTemplates": {
        "test" : "component-templates/TemplateName.test.js"
      },
      "path": "src/components",
      "withStyle": true,
      "withTest": true,
      "withStory": true,
      "withLazy": true
    }
  }
}
```

**Comment by arminbro** [comment]
Good catch, @BudWhiteStudying. Nicely done using a custom test template as a workaround until I could address the issue.

This should now be resolved in version 8.4.9.

Thanks again for bringing this to my attention, and please feel free to reach out if you have any further questions.
